mockrxandroidble/src/main/java/com/polidea/rxandroidble/mockrxandroidble/RxBleClientMock.java

```diff
@@ -271,18 +277,25 @@ public RxBleDevice getBleDevice(@NonNull String macAddress) {
         return createScanOperation(filterServiceUUIDs);
     }
 
+    public void simulateDeviceDiscovery(@NonNull RxBleDeviceMock deviceMock) {
```

Comment:
That's a good idea :) What I'd do is to provide an observable to the mock. In the builder there would be two options:

- public Builder addDevice(@NonNull RxBleDevice rxBleDevice)
- public Builder setDeviceDiscoveryObservable(@NonNull Observable<RxBleDevice> rxBleDevice)

Then the builder with pass merged data feed to the mock. This would be really flexible and easy to use.